mono/metadata/icall.c

```diff
@@ -1403,10 +1452,12 @@ ves_icall_System_Type_internal_from_name (MonoString *name,
 	/* mono_reflection_parse_type() mangles the string */
 	if (!parsedOk) {
 		mono_reflection_free_type_info (&info);
-		g_free (str);
 		if (throwOnError) {
-			mono_set_pending_exception (mono_get_exception_argument("typeName", "failed parse"));
+			mono_error_init (&error);
```

Comment:
Oh if you mean do I need to do `mono_error_init`, then the answer is yes: nothing else initializes the `flags` field.